cdc/generation.cc

```diff
@@ -345,39 +345,400 @@ static void do_update_streams_description(
 
     std::vector<cdc::stream_id> streams_vec(streams_set.begin(), streams_set.end());
 
-    sys_dist_ks.create_cdc_desc(streams_ts, streams_vec, ctx).get();
+    co_await sys_dist_ks.create_cdc_desc(streams_ts, streams_vec, ctx);
     cdc_log.info("CDC description table successfully updated with generation {}.", streams_ts);
 }
 
-void update_streams_description(
+future<> update_streams_description(
         db_clock::time_point streams_ts,
         shared_ptr<db::system_distributed_keyspace> sys_dist_ks,
         noncopyable_function<unsigned()> get_num_token_owners,
         abort_source& abort_src) {
     try {
-        do_update_streams_description(streams_ts, *sys_dist_ks, { get_num_token_owners() });
-    } catch(...) {
+        co_await do_update_streams_description(streams_ts, *sys_dist_ks, { get_num_token_owners() });
+    } catch (...) {
         cdc_log.warn(
             "Could not update CDC description table with generation {}: {}. Will retry in the background.",
             streams_ts, std::current_exception());
 
         // It is safe to discard this future: we keep system distributed keyspace alive.
-        (void)seastar::async([
-            streams_ts, sys_dist_ks, get_num_token_owners = std::move(get_num_token_owners), &abort_src
-        ] {
+        (void)(([
+            _streams_ts = streams_ts,
+            _sys_dist_ks = std::move(sys_dist_ks),
+            _get_num_token_owners = std::move(get_num_token_owners),
+            &_abort_src = abort_src
+        ] () mutable -> future<> {
+            auto streams_ts = _streams_ts;
+            auto sys_dist_ks = std::move(_sys_dist_ks);
+            auto get_num_token_owners = std::move(_get_num_token_owners);
+            auto& abort_src = _abort_src;
+
             while (true) {
-                sleep_abortable(std::chrono::seconds(60), abort_src).get();
+                co_await sleep_abortable(std::chrono::seconds(60), abort_src);
```

Comment:
Won't this cause an ugly "exceptional future ignored" warning to be printed when Ctrl+C is pressed? AFAIK, aborted sleep causes an exception to be thrown.

Reply:
Yes, but that happens on node shutdown, I don't know if catching this exception is even worth it.